project_utils/submodule.py

```diff
@@ -148,6 +156,33 @@ def _get_submodule_enable(manifest, remote_name, combo):
             return source.enable_submodule
     return False
 
+def _get_blobless_enable(manifest, remote_name, combo):

```

Comment:
These two methods will only take into account repositories which have manifest level filtering set and will not catch cases of users inputing a filter type on the clone cmd line. This could lead to a mismatch in clone types for the parent repo and the submodule containing them. 

In the case where the parent repository is not already present on the system we must also check command line parameters. (Note: in some situations, no selective initialization, the initialization is performed in the _update() method)

In the case where the repository is already present we should defer to the git config file for that repository. Specifically the command git config remote.origin.partialclonefilter will return the partial clone type if any that was used to create the repository.